.github/workflows/sdk-publish.yaml

```
@@ -377,13 +376,14 @@ jobs:
     steps:
       - name: Tune GitHub-hosted runner network
         uses: smorimoto/tune-github-hosted-runner-network@v1
+      - uses: actions/checkout@v3
```

These should at least be using v4, and for security reasons, pinning SHA:

```suggestion
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```